build_scripts/desktop/get_variant.sh

```diff
@@ -0,0 +1,185 @@
+#!/bin/bash
+set -e
+
+# Given a filename on the command-line, figure out what variant it should be packaged into.
+# For example, firebase-cpp-sdk-windows-x64-Release-static-build.tgz would be:
+#   OS: Windows
+#   Compiler: VS2015 (default for Windows)
+#   CPU: x64
+#   Debug mode: Release
+#   CRT linkage: MT ("static")
+# So the build variant would be: VS2015/MT/x64/Release
+
+path=$1
+
+if [[ -z "${path}" ]]; then
+    echo "Please specify package file." 1>&2
+    exit 2
+fi
+
+filename=$(basename "${path}")
+
+# Ensure the filename starts with firebase-cpp-sdk-
+if [[ "${filename}" != "firebase-cpp-sdk-"* ]]; then
+    echo "Can only guess variant for firebase-cpp-sdk files. ${filename}" 1>&2
+    exit 2
+fi
+
+os=
+arch=
+arch_win=
+debugmode=Release
+linkage=MD
```

Comment:
Better name for this variable would be `msvc_runtime_library` or `msvc_crt_linkage` just to make it clear that it makes sense only for msvc.

Reply:
Done.